### PR TITLE
Mobile nav

### PR DESCRIPTION
**Pull request by marco-salcedo**
# Description

This PR adds the Mobile Nav component to main branch.

# Links

Related pull requests: https://github.com/Hawaian-United-Okinawa-Association/huoa-react/pull/71

Design: https://www.figma.com/proto/MRix6Z1eMDEfeUDjP97e5m/Nav-Bar-Samples?node-id=86%3A149&scaling=min-zoom

# Additional Info

Preview:

### CODE REVIEW

File: src/reducers/scrollReducer.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const INITIAL_STATE = true;
+
+export default (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case TOGGLE_SCROLL:
+      return !state;
+    default:
+      return state;
+  }
+};
```
Comment (mattlaguardia) [inline src/reducers/scrollReducer.js:12]: This functionality seems weird to me, example this switch statement's cases. Think about this a bit and what the case statements should be checking.

File: src/components/NavMobile/NavMobile.scss [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+        height: 1px;
+        min-height: 1px;
+        border: 1px;
+        background-color: $primary-1;
+      }
+
+      &--footer {
+        display: flex;
+        flex-direction: column;
+        padding: 1rem 0 3rem 0;
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.scss:225]: Short hand

```suggestion
        padding: 1rem 0 3rem;
```

File: src/components/NavMobile/NavMobile.scss [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+            0% {
+              height: 0;
+              transform: scaleY(0);
+            }
+
+            100% {
+              height: 5rem;
+              transform: scaleY(1);
+            }
+          }
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.scss:176]: We should list keyframes at bottom of stylesheet

File: src/components/NavMobile/NavMobile.scss [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+          animation: derotateCarrot 200ms ease-in-out forwards;
+        }
+      }
+
+      &--dropdown {
+        display: none;
+        margin: 0 -2rem;
+        padding: 0;
+
+        &--active {
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.scss:165]: hmm mike's linter doesnt throw an error but this is pretty nested, it should be throwing an error saying this is too nested. ie this is what the class is going to look like: `.nav-mobile__sidebar--dropdown--active`

File: src/components/NavMobile/NavMobile.scss [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+            100% {
+              transform: rotate(0deg);
+            }
+          }
+
+          margin-right: 1rem;
+          animation: derotateCarrot 200ms ease-in-out forwards;
+        }
+      }
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.scss:158]: see comment of being too nested

File: src/components/NavMobile/NavMobile.scss [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+      &--active {
+        @keyframes slideIn {
+          from {
+            right: -35rem;
+          }
+
+          to {
+            right: 0;
+          }
+        }
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.scss:80]: place all keyframes at the end of the file

File: src/components/NavMobile/NavMobile.scss [hunk 7]
```diff
@@ -0,0 +1,332 @@
+@import '../../styles//utilities/colors.scss';
+@import '../../styles/utilities/breakpoints.scss';
+@import '../../styles/utilities/fonts.scss';
+
+.nav-mobile {
+  -webkit-tap-highlight-color: transparent;
+  display: none;
+  overflow: hidden;
+}
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.scss:9]: You declare this in the next class declaration in the media query

```suggestion
.nav-mobile {
  -webkit-tap-highlight-color: transparent;
  display: none;
}
```

File: src/components/NavMobile/NavMobile.js [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+  };
+
+  useEffect(() => {
+    function handleResize() {
+      setScreenWidth(window.innerWidth)
+      if(screenWidth >= 1200){
+        resetSideNav('init');
+      }
+    }
+    window.addEventListener('resize', handleResize)
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.js:37]: Use es6 arrow function please

```
    function handleResize() {
      setScreenWidth(window.innerWidth)
      if(screenWidth >= 1200){
        resetSideNav('init');
      }
    }
    window.addEventListener('resize', handleResize)
```

File: src/components/NavMobile/NavMobile.js [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+  })
+
+  const handleAnimation = (hook) => {
+    switch(hook){
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.js:41]:
```suggestion
    switch (hook) {
```

File: src/components/NavMobile/NavMobile.js [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+        return '--inactive';
+      case true:
+        return '--active';
+      default:
+        return null;
+    }
+  };
+
+  const handleScroll = () => {
+    if(window.screen.width < 768) dispatch(toggleScroll());
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.js:54]: Is eslint not throwing console errors for these?

```suggestion
    if (window.screen.width < 768) dispatch(toggleScroll());
```

File: src/components/NavMobile/NavMobile.js [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+      return (
+        <li className="nav-mobile__listItem" key={ item.slug } >
+          <Link onClick={() => { resetSideNav(false); handleScroll(); }} to={`/${item.slug}`} className="nav-mobile__sidebar--dropdown-item">{item.title}</Link>
+        </li>
+      );
+    });
+  };
+
+  const standardItem = (item) => {
+    if(item.children.length){
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.js:68]:
```suggestion
    if (item.children.length) {
```

File: src/components/NavMobile/NavMobile.js [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+      const handleDrop = (title) => {
+        if(title === closeDropdown){
+          return false;
+        }else if(title === dropdownOpen){
+          return true;
+        }else{
+          return 'init'
+        }
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.js:76]:
```suggestion
        if (title === closeDropdown) {
          return false;
        } else if (title === dropdownOpen) {
          return true;
        } else {
          return 'init'
        }
```

File: src/components/NavMobile/NavMobile.js [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+            {item.title}
+            <Carrot className={`nav-mobile__sidebar--carrot${handleAnimation(handleDrop(item.title))}`} />
+          </div>
+          <ul className={`nav-mobile__sidebar--dropdown${handleAnimation(handleDrop(item.title))}`}>
+            { renderDropdown(item.children) }
+          </ul>
+        </li>
+      )
+    } else {
+      return(
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.js:95]:
```suggestion
      return (
```

File: src/components/NavMobile/NavMobile.js [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+              {item.title}
+            </div>
+          </Link>
+        </li>
+      )
+    }
+  };
+
+  const redirectItem = (item) => {
+    return(
```
Comment (mattlaguardia) [inline src/components/NavMobile/NavMobile.js:108]:
```suggestion
    return (
```

File: src/reducers/scrollReducer.js (hunk 1 above)
Comment (marco-salcedo) [reply src/reducers/scrollReducer.js:12]: Left this unfixed for a future dev kaizen

**Pull request merged**